pkg/query/storeset.go

```diff
@@ -375,12 +386,14 @@ func (s *StoreSet) updateStoreStatus(store *storeRef, err error) {
 
 	status.LastError = err
 	status.LastCheck = time.Now()
+	s.storeStatus.WithLabelValues(store.addr).Set(0)
```

Comment:
IMO I would update this metric in `func (s *StoreSet) Update(ctx context.Context) {` - it would be much more explicit.

Reply:
In the Update method, there are a lot of blocks that calls the `updateStoreStatus` method. Setting the store status metric literally needs to happen every time `updateStoreStatus` is called in `Update`. 

So with this in mind, it made more sense to place this piece of code in the `updateStoreStatus` method as this metric literally represent the status that is being updated in this method. 

Placing it in the Update method also means that the value needs to be set manually and not via logic. E.g. 

```
s.updateStoreStatus(store, nil)
s.storeStatus.WithLabelValues(store.addr).Set(1)

s.updateStoreStatus(store, someError)
s.storeStatus.WithLabelValues(store.addr).Set(0)
```

Instead of based on logic:

```
	s.storeStatus.WithLabelValues(store.addr).Set(0)

	if err == nil {
		s.storeStatus.WithLabelValues(store.addr).Set(1)
	}

```

With this in mind, do you still think It needs to go in the Update method?